File: Javascript/BinaryTreeInorderTraversal.js
/**
 * Problem: https://leetcode.com/problems/binary-tree-inorder-traversal/
 * Author: Dung Nguyen Tien (Chris)
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    var res = [];
    
    // left => node.val => right
    trace(root, res);
    return res;
};

function trace(root, res) {
    if (root == null) return;
    
    if (root.left != null) trace(root.left, res);
    
    res.push(root.val);

    if (root.right != null) trace(root.right, res);
}

File: C#/BinaryTreeInorderTraversal.cs
/**
 * Problem: https://leetcode.com/problems/binary-tree-inorder-traversal/
 * Author: Dung Nguyen Tien (Chris)
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<int> InorderTraversal(TreeNode root) {
        var res = new List<int>();
        
        InorderTraversal(root, ref res);
        return res;
    }
    
    public void InorderTraversal(TreeNode node, ref List<int> res) {
        if (node == null) return;
        
        InorderTraversal(node.left, ref res);
        res.Add(node.val);
        InorderTraversal(node.right, ref res);
    }
}